core/src/main/java/org/apache/iceberg/RowLevelOperationMode.java

```diff
@@ -43,9 +43,9 @@ public enum RowLevelOperationMode {
 
   public static RowLevelOperationMode fromName(String modeName) {
```

Comment:
Would it be cleaner to add the constant to `RowLevelOperationMode` instead?

```
public enum RowLevelOperationMode {
  COPY_ON_WRITE("copy-on-write"),
  MERGE_ON_READ("merge-on-read");

  private final String modeName;

  RowLevelOperationMode(String modeName) {
    this.modeName = modeName;
  }

  public static RowLevelOperationMode fromName(String modeName) {
    ...
  }

  public String modeName() {
    return modeName;
  }
```

We can then use it in `TableProperties` and other places.

```
public static final String MERGE_MODE = "write.merge.mode";
public static final String MERGE_MODE_DEFAULT = RowLevelOperationMode.COPY_ON_WRITE.modeName();
```

Reply:
+1. Rather than putting them in TableProperties, i prefer to keep them separate.

Reply:
+1 to that idea

Reply:
Good idea. Done